### ISSUE TITLE
[META] Blockers for merging into core

### ISSUE THREAD

**Issue by jenlampton** [backdrop-contrib/reference#16]
Here are some blockers: (Thanks @herbdool!!!)

- [ ]  #11
- [ ]  #15 
- [ ]  #22
- [ ]  #27
- [x]  #21
- [x]  #23
- [x]  #19

---

original issue:  I have a few sites that need a reference module, and would love to take this for a spin and kick the tires, but I notice there's not a release yet (it's not on backdropcms.org).

Do we have a list of what the release blockers are? I'd love to help sort those out rather than continue using references (which has a stable release).

I notice that the todo list is quite long: https://github.com/backdrop-contrib/reference/issues/4

**Comment by klonos** [comment]
I would like to grab the chance to note that I find the introduction of experimental modules in D8 core very effective and I believe that we should introduce the same practice in Backdrop core for core candidate projects. Perhaps a matter worth bringing up during our next dev meeting.

**Comment by olafgrabienski** [comment]
I'm also interested to use the module on two sites soon and would prefer to use a 'regular' release. I watched the last dev meeting and understood that there will be such a release shortly. @mikemccaffrey Are there still blockers where we can help in the meantime?

**Comment by klonos** [comment]
Yep, missing tests were mentioned, and although as a matter of principle all projects should have full test coverage, I feel that "full" can never be achieved. I think @jenlampton has mentioned before that "perfect is the enemy of done" 😄 So, @mikemccaffrey, can you please officially list here what is holding you from releasing even an alpha version? ...perhaps there can be things where we can help.

**Comment by docwilmot** [comment]
To remind, there is a todo list here: https://github.com/backdrop-contrib/reference/issues/4 with lots of unchecked boxes
Maybe thats the problem?

**Comment by mikemccaffrey** [comment]
Well, I was more looking for other people to tell me when the module had enough functionality for a release. However, since the folks who have tested it so far have not reported any major projects, it seems like we are good for a beta release, which I just pushed: https://github.com/backdrop-contrib/reference/releases/tag/1.x-1.0-beta1

What do we need to do to get it listed on backdropcms.org?

**Comment by jenlampton** [comment]
> What do we need to do to get it listed on backdropcms.org?

You need a stable release (one without -alpha or -beta at the end)

**Comment by jenlampton** [comment]
Oh, I think there is also a problem with the packager right now...  I had that issue the other day too.  @Gormartsen any progress on that one?

**Comment by herbdool** [comment]
Here are some blockers before it gets put into core and before it upgrades References fields to fields for this module. *All* of these are existing features of References that would break if sites could no longer use References:

* Allow a view to be used to filter content that can be referenced https://github.com/backdrop-contrib/reference/issues/23
* Add more widget types https://github.com/backdrop-contrib/reference/issues/22
* Add Views relationship: Referencing field https://github.com/backdrop-contrib/reference/issues/21
* Hooks so that we can have modules like Node Reference URL Widget, Reference Prepopulate, (Node) Reference Dialog https://github.com/backdrop-contrib/reference/issues/19
* Need to check access when retrieving entities for autocomplete reference field https://github.com/backdrop-contrib/reference/issues/15
* Needs to check access when displaying referenced entities https://github.com/backdrop-contrib/reference/issues/11

And some others that are bugs or UX issues:

* Manage display: Check defaults for "Rendered entity" https://github.com/backdrop-contrib/reference/issues/14
* Tests for all functionality

**Comment by jenlampton** [comment]
@herbdool that's a fantastic start to a list, and all are manageable. I expect we may find some other things as we go, and we can add them to this META as needed :)

**Comment by mikemccaffrey** [comment]
> Here are some blockers before it gets put into core and before it upgrades References fields to fields for this module...

Many of these items seem outside the scope of what is needed for an initial contrib release. Should we perhaps change the title of this ticket to describe that these are the blockers keeping the module out of core?
